app/controllers/membership_activation_tokens_controller.rb

```diff
@@ -26,17 +27,19 @@ def submit
         return
       end
 
-      @user.add_role :member
+      @user.activate
       @user.touch(:consented)
+
       sign_out
 
       @membership_activation_token.destroy
 
-      flash[:success] = 'You have successfully (re)-activated your account! Please log in to continue.'
+      helpers.append_to_flash(:success, 'You have successfully (re)-activated your account! Please log in to continue.')
 
       redirect_to admin_url
     else
-      flash[:error] = 'You need to give consent before you can create an account.'
+      helpers.append_to_flash(:error, 'You need to give consent before you can create an account.')
```

Comment:
again, i'd suggest something like: You need to agree to T&Cs/Privacy Policy - you need to give consent sounds very affacing